debugger/symbols.ml

```diff
@@ -120,36 +121,68 @@ let read_symbols bytecode_file =
           in
           let sorted_evl = List.sort cmp evl in
           modules := md :: !modules;
-          Hashtbl.add all_events_by_module md sorted_evl;
+          Hashtbl.add all_events_by_module md (frag, sorted_evl);
           let real_evl =
             List.filter
               (function
                  {ev_kind = Event_pseudo} -> false
                | _                        -> true)
               sorted_evl
           in
-          Hashtbl.add events_by_module md (Array.of_list real_evl))
+          Hashtbl.add events_by_module md (frag, Array.of_list real_evl))
     all_events
 
+let read_symbols frag bytecode_file =
+  let all_events, all_dirs = read_symbols' bytecode_file in
+  program_source_dirs := !program_source_dirs @ (String.Set.elements all_dirs);
+  add_symbols frag all_events
+
+let erase_symbols frag =
```

Comment:
Shouldn't this function also update the `events` reference?

Reply:
The `events` reference was not used anywhere. I dropped it.